packages/filler/src/tests/filler.test.ts

```diff
@@ -150,45 +152,106 @@ describe.sequential("Basic", () => {
-				resolve(data.orderId)
+			eventMonitor.on("orderFilled", (data: { orderId: string; hash: string }) => {
+				console.log("Order filled by event monitor:", data.orderId, data.hash)
+				resolve(data)
 			})
 		})
 
-		const orderFilledId = await orderFilledPromise
-		console.log("Order filled:", orderFilledId)
+		const { orderId, hash: filledHash } = await orderFilledPromise
+		console.log("Order filled:", orderId, filledHash)
+
+		const filledReceipt = await gnosisChiadoPublicClient.waitForTransactionReceipt({
+			hash: filledHash as `0x${string}`,
+			confirmations: 1,
+		})
 
 		let isFilled = await checkIfOrderFilled(
-			orderFilledId as HexString,
+			orderId as HexString,
 			gnosisChiadoPublicClient,
 			gnosisChiadoIntentGateway.address,
 		)
 
 		expect(isFilled).toBe(true)
 
-		console.log("Checking if order is filled at the source chain...")
-		await new Promise((resolve) => setTimeout(resolve, 60 * 1000))
-
-		isFilled = await checkIfOrderFilled(orderFilledId as HexString, bscPublicClient, bscIntentGateway.address)
-		let maxAttempts = 20
-		while (!isFilled && maxAttempts > 0) {
-			console.log("Order not filled at the source chain, retrying storage check in 30 seconds...")
-			console.log("Max storage checks left:", maxAttempts)
-			await new Promise((resolve) => setTimeout(resolve, 30 * 1000))
-			isFilled = await checkIfOrderFilled(orderFilledId as HexString, bscPublicClient, bscIntentGateway.address)
-			maxAttempts--
+		// parse EvmHost PostRequestEvent emitted in the transcation logs
+		const event = parseEventLogs({ abi: EVM_HOST, logs: filledReceipt.logs })[0]
+
+		if (event.eventName !== "PostRequestEvent") {
+			throw new Error("Unexpected Event type")
 		}
 
-		expect(isFilled).toBe(true)
+		const request = event.args
+		console.log("PostRequestEvent", { request })
+		const commitment = postRequestCommitment(request).commitment
+
+		for await (const status of indexer.postRequestStatusStream(commitment)) {
+			console.log(JSON.stringify(status, null, 4))
+			switch (status.status) {
+				case RequestStatus.SOURCE_FINALIZED: {
+					console.log(
+						`Status ${status.status}, Transaction: https://gargantua.statescan.io/#/extrinsics/${status.metadata.transactionHash}`,
+					)
+					break
+				}
+				case RequestStatus.HYPERBRIDGE_DELIVERED: {
+					console.log(
+						`Status ${status.status}, Transaction: https://gargantua.statescan.io/#/extrinsics/${status.metadata.transactionHash}`,
+					)
+					break
+				}
+				case RequestStatus.HYPERBRIDGE_FINALIZED: {
+					console.log(
+						`Status ${status.status}, Transaction: https://gnosis-chiado.blockscout.com/tx/${status.metadata.transactionHash}`,
+					)
+					const { args, functionName } = decodeFunctionData({
+						abi: HandlerV1_ABI,
+						data: status.metadata.calldata,
+					})
+
+					expect(functionName).toBe("handlePostRequests")
+
+					try {
+						const hash = await bscHandler.write.handlePostRequests(args as any, {
```

Comment:
I have now removed storage checks for filled status and now using the indexer to stream and self relay. But at this like when the relayer tried to handlePostRequest in the sourceChain, it get's a weird error: hostFrozen.

Reply:
You don't need to submit it, the relayer  will do that, just wait for DESTINATION status.